### ISSUE TITLE
Add support for archiving GitHub mirrors automatically based on "archived" API parameter on the remote repo

### ISSUE THREAD

**Issue by kbdharun** [go-gitea/gitea#27909]
### Feature Description

Currently, there isn't an archive option for mirror repositories (and it is available for only regular ones as far as I can remember).

It would be great if Gitea could read the "archived" parameter from the GitHub mirror's remote API configuration during sync. If the archive value is set to true then the mirror can be archived automatically.

Once the mirror is archived, we can continue to query the API alone (while disabling other sync operations) to check if the repo has been unarchived. If the repo has been unarchived, the mirror can also be unarchived (and repo sync can be done as set).

----

## Examples

1. [Example archived repo](https://github.com/kbdharun/Java-Programs) with archived parameter set to "true" in the [API](https://api.github.com/repos/kbdharun/Java-Programs).
2. [Example normal repo](https://github.com/tldr-pages/tldr) with archived parameter set to "false" in the [API](https://api.github.com/repos/tldr-pages/tldr).

----

**Comment by KN4CK3R** [comment]
I don't think this will work reliable. The default mirror interval is 8h. If you unarchive, change files and re-archive a repo on Github inbetween this time slot, Gitea never sees a change.

And after the migration we don't know the source type of the mirror anymore because only the git remote data is needed.
